docs/src/submodules/Utilities/overview.md

```diff
@@ -149,11 +151,11 @@ julia> model = MOI.Utilities.CachingOptimizer(
            MOI.Utilities.Model{Float64}(),
            PathOptimizer{Float64}(),
        )
-MOIU.CachingOptimizer{PathOptimizer{Float64},MOIU.Model{Float64}}
+MOIU.CachingOptimizer{MOIU.GenericOptimizer{Float64,MOIU.VectorOfConstraints{MOI.VectorAffineFunction{Float64},MOI.Complements}},MOIU.GenericModel{Float64,MOIU.ModelFunctionConstraints{Float64}}}
```

Comment:
Can we overload `show` for this model so that it prints `PathOptimizer{Float64}`? this issue going to cause the JuMP model printing to be horrible.

Reply:
Isn't it printing the type `PathOptimizer`, not an object of type `PathOptimizer` So we should overload `show(:IO, ::Type{PathOptimizer{T}})`?

Reply:
Oh, yeah.
```Julia
Base.show(io::IO, ::Type{$(name){T,C}}) where {T,C} = print(io, "$(name){T}")
```

Reply:
> this issue going to cause the JuMP model printing to be horrible.

Which printing ? JuMP does not print the type of the backend

Reply:
Anyone interacting with the backend:
https://jump.dev/JuMP.jl/dev/manual/models/#Backends
Or anyone interacting with MOI models like https://github.com/chkwon/PATHSolver.jl which are based on this.

Reply:
If we define this show method then how can the user see what's the actual type?
We also have `const AffExpr =...` and we don't define printing for this type

Reply:
I withdraw my objection. We can always revisit if it becomes a pain, or down-stream solvers can define their own `show`.